lib/src/error_handler_delegate.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:dio/dio.dart';
+
+/// Delegate that provides a set methods for controlling
+/// the process of error handling through all requests
+abstract class ErrorHandlerDelegate {
+
+  /// This method will be called for error handling purposes
+  void onErrorCaught(DioError error);
+
+  /// Calls to determine if the request failed
+  /// and we can proceed it
+  bool isErrorCaught(DioError error);
```

Comment:
Maybe we should rename it as `canHandleError` and `onErrorCaught` => `handleError`

Reply:
Fixed